backend/core/src/user/user.service.ts

```diff
@@ -79,20 +81,47 @@ export class UserService {
     return savedPasswordHash === verifyPasswordHash
   }
 
+  public async confirm(dto: ConfirmDto) {
+    const user = await this.repo.findOne({ confirmationToken: dto.token })
+    if (!user) {
+      throw new HttpException(USER_ERRORS.TOKEN_MISSING.message, USER_ERRORS.TOKEN_MISSING.status)
+    }
+    user.confirmedAt = moment().toDate()
+    user.confirmationToken = null
+
+    try {
+      await this.repo.save(user)
+      return user
+    } catch (err) {
+      throw new HttpException(USER_ERRORS.EMAIL_IN_USE.message, USER_ERRORS.EMAIL_IN_USE.status)
+    }
+  }
+
   public async createUser(dto: UserCreateDto) {
+    let user = await this.findByEmail(dto.email)
+    if (user) {
+      if (!user.confirmedAt) {
+        return user
```

Comment:
This is only in case the account isn't confirmed, so we can resend email confirmation. We can either change controller to return text like `ok` or create a separate controller action to resend email confirmation. Which do you feel like is better?

Reply:
@software-project I think we will need resending a confirmation email anyway, so maybe let's go that way. I think resending should be for admins only right?

Reply:
@pbn4 nope, it's part of that ticket to allow user to send email confirmartion again. There is a button to do it, but also if user just fills out the form again the way I had it prepared will reassure that account is not created twice and just email confirmation will be sent. So IMHO we just should restrict whats returned by the controller.